packages/opentelemetry-api/src/api/global-utils.ts

```diff
@@ -28,11 +29,16 @@ export const GLOBAL_PROPAGATION_API_KEY = Symbol.for(
 );
 export const GLOBAL_TRACE_API_KEY = Symbol.for('io.opentelemetry.js.api.trace');
 
+export const GLOBAL_DIAG_LOGGER_API_KEY = Symbol.for(
+  'io.opentelemetry.js.api.diag'
+);
+
 type Get<T> = (version: number) => T;
 type OtelGlobal = Partial<{
   [GLOBAL_CONTEXT_MANAGER_API_KEY]: Get<ContextManager>;
   [GLOBAL_PROPAGATION_API_KEY]: Get<TextMapPropagator>;
   [GLOBAL_TRACE_API_KEY]: Get<TracerProvider>;
+  [GLOBAL_DIAG_LOGGER_API_KEY]: Get<DiagAPI>;
```

Comment:
Created the DiagAPI as the global instance, although this could have some unexpected side effects (which I think is also true for context, propagation and trace.

Specific Scenario:
An application includes multiple "components" with each component importing and using OT... As these are registered on the global instance there will be unexpected interactions between the different components usage of the global API's. This is probably mostly an issue for browser based scenario's vs server side node based etc.

The solution to the above would be to introduce some form of API context that is used and passed around to all API operations etc.

Reply:
Which it will as when DiagAPI registers itself it supplies a NoopDiagAPI fallback.

Reply:
But the scenario I'm pointing out is that you could have many components using the SAME version of the API, so the Noop doesn't come into play here as everything is registered as a global.

Reply:
Hmm I wonder if a simpler solution would be to prevent different signals from using different API versions. This is what is proposed in #1772